openedx/adg/lms/webinars/views.py

```diff
@@ -56,7 +58,12 @@ def post(self, request, pk, action):
             WebinarRegistration.objects.update_or_create(
                 webinar=self.object, user=user, defaults={'is_registered': is_registering}
             )
-            # pylint: disable=expression-not-assigned
-            send_webinar_registration_email(self.object, user.email) if is_registering else None
+            if is_registering:
+                send_webinar_registration_email(self.object, user.email)
+                schedule_webinar_reminders(user.email, self.object)
+            else:
+                registration = WebinarRegistration.objects.get(webinar=self.object, user=user)
+                task_cancel_mandrill_email.delay(registration.starting_soon_mandrill_reminder_id)
+                task_cancel_mandrill_email.delay(registration.week_before_mandrill_reminder_id)
```

Comment:
There is no need for line # 65, you can get registration from `update_or_create` like this

```
registration, created = WebinarRegistration.objects.update_or_create(
                webinar=self.object, user=user, defaults={'is_registered': is_registering}
            )
```

Also:

```
if not created:
    task_cancel_mandrill_email.delay(registration.starting_soon_mandrill_reminder_id)
    task_cancel_mandrill_email.delay(registration.week_before_mandrill_reminder_id)
```
If Webinar Registration entry is just created for cancelation, then it means there were no prior scheduled emails